server/autotest_server.py

```diff
@@ -353,6 +354,7 @@ def setup_files(files_path, tests_path, test_scripts, hooks_script,
         if fd == 'f' and os.path.relpath(file_or_dir, tests_path) not in test_scripts:
             permissions -= 0o111
         os.chmod(file_or_dir, permissions)
+    os.chdir(tests_path)
```

Comment:
I think the `os.chdir` should be done (and then undone) in the `run_hooks` function using a context manager. This will make sure that only the hooks scripts are run in the `tests_path` directory. Changing it only once and then leaving it means that most of the server code will be run with `pwd==tests_path` which is currently not a problem but may be confusing for future changes